test/persistence-hooks.suite.js

```diff
@@ -2946,6 +2946,252 @@ module.exports = function(dataSource, should, connectorCapabilities) {
 
     });
 
+    describe('PersistedModel.upsertWithWhere', function() {
+      var createdId = null;
+      var createdExtra = null;
+      var createdName = null;
+
+      it('triggers hooks in the correct order on create', function(done) {
+        monitorHookExecution();
+        TestModel.upsertWithWhere({ where: { extra: 'e0101' }},
+          { id: 'not-found', name: 'not found' },
+          function(err, record, created) {
+            if (err) return done(err);
+            createdId = record.id;
+            createdExtra = record.extra;
+            createdName = record.name;
+            hookMonitor.names.should.eql([
+              'access',
+              'before save',
+              'persist',
+              'loaded',
+              'after save',
+            ]);
+            done();
+          });
+      });
+
+      it('triggers hooks in the correct order on update', function(done) {
+        monitorHookExecution();
+        TestModel.upsertWithWhere({ where: { extra: 'e0101' }},
```

Comment:
How comes `{extra: 'e0101' } ` did not match any records in the previous test ("... on create"), but it matches an existing record now in this test ("... on update")?